pkg/render/node.go

```diff
@@ -1433,6 +1472,42 @@ func (c *nodeComponent) nodePodSecurityPolicy() *policyv1beta1.PodSecurityPolicy
 	return psp
 }
 
+// hostPathInitContainer creates an init container that changes the permissions on hostPath volumes
+// so that they can be written to by a non-root container.
+func (c *nodeComponent) hostPathInitContainer() corev1.Container {
+	rootUID := int64(0)
+	mounts := []corev1.VolumeMount{
+		{
+			MountPath: "/var/run",
+			Name:      "var-run",
+			ReadOnly:  false,
+		},
+		{
+			MountPath: "/var/lib",
+			Name:      "var-lib",
+			ReadOnly:  false,
+		},
+	}
+
+	return corev1.Container{
+		Name:         "hostpath-init",
+		Image:        "busybox:latest",
```

Comment:
I'm not sure but I think this image should go through `ResolveImages` so that the user can use their own image registry. But that might mean that we need to put this image under the calico "namespace" like `quay.io/calico/busybox`.

Also can we use a specific image tag so that we can control when the version is bumped?

Reply:
You're right I need to see what images we have available for this. I think we should have something we maintain for small tasks like this, but if not, I can bake something into calico-node and have that do these commands for me.

Reply:
This should be resolved with https://github.com/projectcalico/node/pull/1238